app/controllers/users_controller.rb

```diff
@@ -228,7 +209,12 @@ def change_status
     # Was the account activated? (do it before User#save clears the change)
     was_activated = (@user.status_change == [User::STATUSES[:registered],
                                              User::STATUSES[:active]])
-    if @user.save
+
+    if params[:activate] && @user.identity_url.nil? && @user.passwords.empty? && @user.auth_source.nil?
+      flash[:error] = I18n.t(:error_status_change_failed,
+                             errors: "User has yet to chose a password or sign up using OpenID Connect.",
```

Comment:
Line is too long. [105/100]<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.